client/getipns.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"context"
+
+	"github.com/ipfs/go-delegated-routing/gen/proto"
+)
+
+func (fp *Client) GetIPNS(ctx context.Context, id []byte) ([][]byte, error) {
+	resps, err := fp.GetIPNSAsync(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	records := [][]byte{}
+	for resp := range resps {
+		if resp.Err != nil {
+			records = append(records, resp.Record)
+		}
+	}
+	if fp.validator != nil {
+		best, err := fp.validator.Select(string(id), records)
```

Comment:
Calling `Select` on an empty set errors
https://github.com/ipfs/go-ipns/blob/5976a80227cc5199414119585cca347bc814647a/record.go#L106.

We should probably return the standard error for a missing record when the set is empty https://github.com/libp2p/go-libp2p-core/blob/02cbdcc4191b0f512c6cf4f2b1a8eeaaf32f1061/routing/routing.go#L15